helperFunctions/subscribeHelpers/InitiateSalesforceSessionHelpers/CheckChatStatusHelper.ts

```diff
@@ -82,6 +83,8 @@ export class CheckChatStatus {
 						const chasitorIdleTimeout = chatEstablishedMessage.chasitorIdleTimeout || false;
 						const sneakPeekEnabled = chatEstablishedMessage.sneakPeekEnabled;
 						const { id, persisantAffinity, persistantKey } = await retrievePersistentTokens(this.read, this.assoc);
+						const agentName = chatEstablishedMessage.name;
+						await extendRoomCustomFields(this.data.room.id,'salesforceAgentName', agentName, this.read, this.modify);
```

Comment:
Instead of setting room customField and then setting message customeField using that roomCustomField on server (https://github.com/WideChat/Rocket.Chat/pull/621/files).

Can we directly assign it to message customField. We are already doing it for  `sneakPeekEnabled`.

Reply:
The API call with the agent's name is only the 'ChatEstablished' message upon escalation and 'ChasitorSessionData' which we don't use. It would be safer for the name to be stored somewhere for the whole session. Alternatively, we can set it up so that as long as the room is active, we make a call for ChasitorSessionData on any refresh.

Reply:
Yes, I got your point we are getting agent name in `ChatEstablished` along with other settings like `sneakPeekEnabled`, `chasitorIdleTimeout`.

What I am saying is to accomplish passing this agent name to LiveChat widget, we don't need to make changes in server code(https://github.com/WideChat/Rocket.Chat/pull/621/files). 

You can set agent name at here https://github.com/WideChat/Apps.Salesforce.LiveAgent/blob/master/helperFunctions/subscribeHelpers/InitiateSalesforceSessionHelpers/CheckChatStatusHelper.ts#L86...L87

`await this.persistence.updateByAssociation(this.assoc,
							{ id, affinityToken: persisantAffinity, key: persistantKey, chasitorIdleTimeout, sneakPeekEnabled , salesforceAgentName});`


At here 
https://github.com/WideChat/Apps.Salesforce.LiveAgent/blob/master/helperFunctions/TimeoutHelper.ts#L17
`const { chasitorIdleTimeout, sneakPeekEnabled, salesforceAgentName } = await retrievePersistentData(read, assoc);`

and here you can directly set it to message customField https://github.com/WideChat/Apps.Salesforce.LiveAgent/blob/master/helperFunctions/TimeoutHelper.ts#L59

`(await msgExtender).addCustomField('salesforceAgentName', salesforceAgentName);`